### PR TITLE
add option deprecation

### PR DESCRIPTION
**Pull request by AllyW**

### CODE REVIEW

**Comment by AllyW** [comment]
image

File: azdev/operations/command_change/util.py [hunk 1]
```diff
@@ -75,6 +75,28 @@ def process_aaz_argument(az_arguments_schema, argument_settings, para):
...
+        opt_type = opt.__class__.__name__
+        if opt_type == "str":
+            option_list.add(opt)
+        elif opt_type == "Deprecated":
+            if hasattr(opt, "hide") and opt.hide:
+                pass
+            if hasattr(opt, "redirect"):
+                option_list.add(opt.redirect)
+            elif hasattr(opt, "target"):
+                option_list.add(opt.target)
```
Comment (kairu-ms) [inline azdev/operations/command_change/util.py]: Why add these two in option list?

Comment (AllyW) [reply azdev/operations/command_change/util.py]: also used as options?

File: azdev/operations/command_change/util.py [hunk 2]
```diff
@@ -99,10 +119,15 @@ def gen_command_meta(command_info, with_help=False, with_example=False):
...
+        if settings.get("action", None):
+            action = settings["action"]
+            if hasattr(action, "__name__") and action.__name__ == "IgnoreAction":
+                # ignore argument like: cmd
+                continue
         para = {
             "name": settings["dest"],
-            "options": sorted(settings["options_list"])
         }
+        process_arg_options(para, settings)
```
Comment (kairu-ms) [inline azdev/operations/command_change/util.py]: `para` seems useless for process_arg_options
```suggestion
        para["options"] = process_arg_options(settings)
```

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**